[RFC] test: Adding shell unit test

rustushki opened the pull request:
It confirms that a non-existent command returns a 127 error code
(showing that system() handles non-existent commands).

Commit:
test: Adding shell unit test

It confirms that a non-existent command returns a 127 error code
(showing that system() handles non-existent commands).

ZyX-I commented:
What’s the point of this? 127 error code has absolutely nothing to do with us, it is a shell convention. More, it is *POSIX* shell convention, `cmd.exe` does not follow it (exits with 9009), neither does `rc` shell (its reimplementation for \*nix systems exits with 1, origin is in Plan 9 and does not run on linux).

justinmk commented:
I don't think we have a test that at least verifies a non-zero result. So perhaps change the test to check for non-zero rather than a specific error code.

ZyX-I commented:
@justinmk Yes, test

      local status, output = os_system('exit 2')
      eq(2, status)

makes much more sense.

Commit:
test: Changing Shell Unit Test

It now tests that a command which is known to return a specific error
code actually does.  Previous implementation would not have worked on
Windows in cmd.exe.

rustushki commented:
I've modified the test to check the return code provided to exit.  Thank you for your feedback.

justinmk commented:
Merged

justinmk commented on test/unit/os/shell_spec.lua:
```diff
@@ -68,5 +68,12 @@ describe('shell functions', function()
+
+    it ('can return non-zero command execution results', function()
+      local ret = 2
+      local cmd = 'exit' .. ' ' .. ret
```
Storing values in variables is good, but it is unnecessary in small scopes like this (where the alternative is just 2 lines). So I removed the variables here.
